File: haskell/Project Euler/Add100_50DigitNums.hs
import IO
import Control.Monad

{-- Project Euler Problem 13 --}

readNIntegers :: Integer -> IO [Integer]
readNIntegers n = do
	if n == 0
		then return []
		else do
			s <- (readLn:: IO Integer)
			rest <- readNIntegers (n-1)
			return (s:rest)

main = do
	numbers <- readNIntegers 100
	let sum = foldr (+) 0 numbers
	let dgts = read (take 10 (show sum))::Integer
	print dgts
	

File: python/Project Euler/Add100_50DigitNums.py
#!/usr//bin/python

''' Project Euler - Problem 13 '''

i = 0
sum = 0
while i<100:
	e = int(input())
	e = str(e)[0:11]
	sum = sum + int(e)
	i = i + 1

s = str(sum)[0:10]
s = int(s)

print s
